akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -657,10 +658,26 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
           (remoteGossip, !remoteGossip.seenByNode(selfUniqueAddress), Newer)
         case _ ⇒
           // conflicting versions, merge
-          (remoteGossip merge localGossip, true, Merge)
+          val prunedLocalGossip = localGossip.members.foldLeft(localGossip) { (g, m) ⇒
```

Comment:
My current understanding is that these prunings (on local and remote) will remove any entry m where
 - the entry is removeUnreachableWithMemberStatus (i.e. is of a status that is removable?)
 - and it is not contained by the other gossip

Now, trying to figure out what happens in various scenarios. Suppose all entries I talk about have removeUnreachableWithMemberStatus true (maybe this is wrong to assume, correct me if wrong)

*Scenario 1*: M is contained in both gossips. In this case it will stay in both pruned versions, and as a result in the merged version

*Scenario 2*: M is contained in only one gossip (lets name it G1, and G2 is the other gossip).
 - G1_pruned will not contain M since G2 does not contain it
 - G2_pruned will contain M since G1 does contain it
 - G1_G2_merged will contain M? How does this work here?